### DISCUSSION TITLE
preferred format for mapcalc strings in python addons?

### DISCUSSION THREAD

**Discussion by CarlosGrohmann** [OSGeo/grass#1958] [Q&A] [answered]
Hi, is there a preferred way to make mapcalc strings in python addons?

Is it ok to use f-strings?

what is better, this:
        `gscript.mapcalc('{a} = (({c}-{b})/({c}+{b}))').format(a=var_a, b=var_b, c=var_c)`

or this?
        `gscript.mapcalc(f'{var_a} = (({var_c}-{var_b})/({var_c}+{var_b}))')`

**Comment by veroandreo** [comment]
AFAIK, `format()` syntax is preferred in general, not only for addons

**Reply by CarlosGrohmann** [reply]
ok thanks

**Reply by wenzeslaus** [reply]
The _format_ function is needed for translatable strings, but otherwise, f-strings are recommend.

**Comment by petrasovaa** [comment, marked as answer]
The f-strings iare preferable, it's more modern syntax. There are some issues for using it with translatable strings, but that's not the case here. I think the format was preferable comparing to the old % syntax.

**Reply by CarlosGrohmann** [reply]
Thanks Anna. I really prefer f-strings

**Reply by wenzeslaus** [reply]
It should be noted that the function also supports string templates with `$` but I believe that was introduced for compatibility with Bash syntax making the transition a little more convenient back in the day, but it does not have any advantages now. Notably, same as the f-string syntax, it does not quote map names automatically which is perhaps the only special thing the _mapcalc_ function could do (but does not).
